pkg/users/account-create-dialog.js

```diff
@@ -228,38 +294,62 @@
         if (field == "locked")
             state.change_passw_force = false;
 
+        if (field == "home_dir") {
+            state.home_dir_dirty = true;
+            state.home_dir_exists = false;
+        }
+
         update();
     }
 
-    function validate(force, real_name, user_name, password, password_confirm) {
+    function validate(force_weak, force_home, real_name, user_name, password, password_confirm) {
         const errs = { };
 
         errs.real_name = validate_real_name(real_name);
-
-        if (password != password_confirm)
-            errs.password_confirm = _("The passwords do not match");
+        errs.password_confirm = validate_password_confirm(password_confirm, password);
 
         if (password.length > 256)
             errs.password = _("Password is longer than 256 characters");
 
         errs.user_name = validate_username(user_name, accounts);
 
-        return password_quality(password, force)
-                .catch(ex => {
-                    errs.password = (ex.message || ex.toString()).replaceAll("\n", " ");
-                })
+        if (!state.home_dir)
+            errs.home_dir = _("No home directory specified");
+
+        const promises = [
+            password_quality(password, force_weak)
+                    .catch(ex => {
+                        errs.password = (ex.message || ex.toString()).replaceAll("\n", " ");
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230317-103941-df531fdc-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)